### PR TITLE
[wasm64] Fix offscreen canvas and offscreen framebuffer

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: system/include/emscripten/threading_legacy.h [hunk 1]
```diff
@@ -49,17 +54,22 @@ typedef struct em_queued_call em_queued_call;
 
 // - starting from the lowest bits upwards, each pair of two subsequent bits
```
Comment (kripken) [inline system/include/emscripten/threading_legacy.h:55]:
```suggestion
// - starting from the lowest bits upwards, each triple of two subsequent bits
```

File: system/include/emscripten/threading_legacy.h [hunk 2]
```diff
@@ -151,6 +181,15 @@ em_queued_call *emscripten_async_waitable_run_in_main_runtime_thread_(EM_FUNC_SI
 // Since we can't validate the function pointer type, allow implicit casting of
 // functions to void* without complaining.
 #define emscripten_sync_run_in_main_runtime_thread(sig, func_ptr, ...) emscripten_sync_run_in_main_runtime_thread_((sig), (void*)(func_ptr),##__VA_ARGS__)
+
+#ifdef __wasm64__
+// For wasm64 we need to special handling of pointer (P) return types since
+// int and pointer have different widths
```
Comment (kripken) [inline system/include/emscripten/threading_legacy.h:187]: Maybe move or duplicate this comment to line 159?

**Review by kripken — APPROVED** [review]

File: system/include/emscripten/threading_legacy.h (hunk 1 above)
Comment (kripken) [reply system/include/emscripten/threading_legacy.h:55]: I just adjusted this comment based on the changes in comments below it. Where is the modified code for this?

Comment (sbc100) [reply system/include/emscripten/threading_legacy.h:55]: Comment update was from a previous experiment.  Reverted this now.

**Pull request merged**